lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -1,13 +1,32 @@
 module ActiveModelSerializers
   module Adapter
     class Attributes < Base
+      def initialize(*)
+        super
+        instance_options[:fieldset] ||= ActiveModel::Serializer::Fieldset.new(fields_to_fieldset(instance_options.delete(:fields)))
+      end
+
       def serializable_hash(options = nil)
         options = serialization_options(options)
         options[:fields] ||= instance_options[:fields]
         serialized_hash = serializer.serializable_hash(instance_options, options, self)
 
         self.class.transform_key_casing!(serialized_hash, instance_options)
       end
+
+      def fields_to_fieldset(fields)
```

Comment:
Because, as per `docs/general/fields.md`, we have a different syntax for fieldset in JSON adapter vs. JSON:API

The test usage in #2100 is `[:title, {comments: [:body]}`. I'm not sure if we ever considered the `fields` option for relationships

However, I'd accept just turning an array into an implicit fieldset, but otherwise require it to be jsonapi-like


> and in a specific controller, you want to return `access_token` only, `fields` will help you:

```ruby
class AnonymousController < ApplicationController
  def create
    render json: User.create(activation_state: 'anonymous'), fields: [:access_token], status: 201
  end
end
```

> Note that this is only valid for the `json` and `attributes` adapter. For the `json_api` adapter, you would use

```ruby
render json: @user, fields: { users: [:access_token] }
```

> Where `users` is the JSONAPI type.